### PR TITLE
ARROW-15816: [R][DOCS] pkgdown config refactoring

### PR DESCRIPTION
**Pull request by eitsupi**
~~Related to ARROW-15734.~~

~~It turns out that we can't upgrade to Bootstrap 5 right away because the arrow's pkgdown site currently uses its own extra.js file.~~
~~So first, staying with the traditional Bootstrap3 now, but make some changes to prepare for upgrading to Bootstrap5 in the future.~~

- ~~Fix articles' title and add setting for article index.~~
- ~~Restore the GitHub icon on the right side of the nav bar.~~

### CODE REVIEW

**Comment by thisisnic** [comment]
@eitsupi This is a bit beyond the scope of what constitutes a minor change - please can you include the issue number in the PR title?

**Comment by eitsupi** [comment]
@thisisnic This PR does not solve ARROW-15734. Do I need to create a new issue?

**Comment by thisisnic** [comment]
> @thisisnic This PR does not solve [ARROW-15734](https://issues.apache.org/jira/browse/ARROW-15734). Do I need to create a new issue?

Yes please

File: r/_pkgdown.yml [hunk 1]
```diff
@@ -40,60 +41,51 @@ navbar:
...
+  - title: Developers
+    navbar: Developers
+    contents:
+      - developing
+      - developers/setup
+      - developers/workflow
+      - developers/debugging
+      - developers/install_details
+      - developers/docker
+      - developers/bindings
```
Comment (thisisnic) [inline r/_pkgdown.yml:87]: There's quite a substantial change here; why was this needed?

Comment (eitsupi) [reply r/_pkgdown.yml:87]: As I mentioned in the body of the PR, this is the way pkgdown recommends to list articles.
https://pkgdown.r-lib.org/reference/build_articles.html#index-and-navbar

The settings before the change will not work with Bootstrap5.
https://github.com/apache/arrow/blob/676b49f4f263a9bdcb7a48e434fa0d52f1da5dcd/r/_pkgdown.yml#L68-L84

**Comment by thisisnic** [comment]
I'm afraid I'm not sure I'm in favour of merging this as it doesn't solve the original problem (activating Bootstrap 5 in order to use the search functionality) and whilst it may provide a partial solution to that, the substantial bit of the work (updating the custom JS) still remains, and there change to the look of the menu is for the worse, in my opinion, sorry.  I will nudge other committers to comment on this though, as they may feel differently.

**Comment by eitsupi** [comment]
Note that if you remove extra.js and activate Bootstrap 5, you will be able to use the search function immediately.

As a user of the arrow package, I have to say that for me, the search function is more necessary than the version switcher.
The current R documentation is not searchable and I often have to search the source code on GitHub to get to the information.
So I'm really hoping to be upgraded to Bootstrap 5 even if it make disable the version switcher.

I think it's really great that you implemented the version switcher, and of course I will follow the decision of the members of this project.
Unfortunately, I don't have the capability to modify extra.js. Please feel free to close this PR.

**Comment by thisisnic** [comment]
Thanks for looking into and I'd agree it'd be great to have both.

**Comment by jonkeane** [comment]
Hello, first I want to say thanks so much for all the work here + other PRs I've seen.

As you both have mentioned: we're a bit stuck and have to choose between supporting search + the version picker we have without putting some time into the JS that does the version picking to be compatible with bootstrap 5. I agree that search is super important, but also that we don't want to lose our version picker since we do that for other languages that Arrow supports.

Would it be possible to push to this branch | PR the changes that get us to using bootstrap 5 (even though it results in a broken site)? Mention the javascript issue, and then we can either leave this open or close it until we have the time to do the necessary javascript work. We'll have a record of what we need to do when we have that javascript skills + headspace to do it, but we don't break anything now. I'm also happy to make sure that you @eitsupi  are listed as a co-author of that PR|commit if + when it merges

**Comment by eitsupi** [comment]
Thank you for checking.And, thank you again for your continued development of this project.

As I mentioned in ARROW-15734, I created this PullRequest in preparation for the upgrade to Bootstrap 5 and intended it to be mergeable at this time.
I apologize for mixing up changes that are essentially related to the upgrade to Bootstrap 5 with other changes.

I would like to change this PR to only restore the GitHub icon and the title of the articles to the correct format, without changing the multi-level list of articles as pointed out.
(It is hard to tell because there is no link from the top page, but some title of the articles are not recognized by pkgdown and the article list is broken. https://arrow.apache.org/docs/6.0/r/articles/)
image

Then, in addition to this PR, I will submit a separate PR that includes all of the current content of this PR and also activates Bootstrap5.

File: r/_pkgdown.yml (hunk 1 above)
Comment (eitsupi) [reply r/_pkgdown.yml]: My explanation above was not adequate.
Correctly, this added setting is for both the index of the article and the nav bar; it is currently overriding the setting for the nav bar, and the overridden setting takes precedence.

**Comment by eitsupi** [comment]
> * With Bootstrap 5, the "index.html" link on projects on the nav bar is always selected when we open the index of the R document, and it doesn't look good, so change it to an absolute path like the other links.

Sorry, I completely misunderstood this to only occur in bs5 but it is the same situation now. (i.e., I recognized it as intentional).
So I have reverted this change.

**Comment by eitsupi** [comment]
I noticed that the multi-level menu in the nav bar cannot be opened from smart phones.

**Comment by eitsupi** [comment]
@thisisnic @jonkeane Is there any reason why this PR cannot be merged? This PR does not break anything.

**Comment by thisisnic** [comment]
Sorry @eitsupi, had not got round to reviewing it again, and had taken your comment

> I noticed that the multi-level menu in the nav bar cannot be opened from smart phones.

to mean that something was broken.  Just to confirm, is that broken on both the live site and in this PR, or something else?

**Comment by eitsupi** [comment]
> to mean that something was broken. Just to confirm, is that broken on both the live site and in this PR, or something else?

Sorry, this was a misunderstanding. If I touched it well, I could select it without any problem.

**Comment by thisisnic** [comment]
> > to mean that something was broken. Just to confirm, is that broken on both the live site and in this PR, or something else?
> 
> Sorry, this was a misunderstanding. If I touched it well, I could select it without any problem.

I'll most likely review this some time after the upcoming release, whenever my focus is on the pkgdown site again. I am about to submit a PR to pkgdown that will incorporate the version selector into pkgdown, and work with Bootstrap 5, so once that's finalised/merged and I make the necessary updates to the Arrow pkgdown site to work with those changes, then I'll take the time to look at it all together.

**Comment by eitsupi** [comment]
If that is the case, I have closed this PR as it seems unnecessary at this time.

**Comment by thisisnic** [comment]
For clarity, I'm not saying your PR isn't helpful - it highlights the changes we'll need to make during the BS5 update. Happy to have it reopened and just pending review/merge at a later point.

**Comment by eitsupi** [comment]
This PR now have nothing to do with BS5, as several modifications have made it different from its original purpose.
Also, #12541 is for the current version of pkgdown, so it is behind the times when using newer pkgdown.

This PR now only restores the currently missing GitHub icon and fixes the article title not being recognized on the article index page.

image

image
